requests/models.py

```diff
@@ -776,23 +776,43 @@ def iter_lines(self, chunk_size=ITER_CHUNK_SIZE, decode_unicode=None, delimiter=
 
         .. note:: This method is not reentrant safe.
         """
-
         pending = None
 
-        for chunk in self.iter_content(chunk_size=chunk_size, decode_unicode=decode_unicode):
+        for chunk in self.iter_content(chunk_size=chunk_size,
+                                       decode_unicode=decode_unicode):
+            # Skip any null responses
+            if not chunk:
+                continue
```

Comment:
Actually I replied too fast: if we want to return the same result on
`['line\r\n']` and  `['line\r\n', '']` (which seems right),
we need to keep this check on top

Otherwise we consume `pending`  and discard it

I think it's safe to discard empty chunks early, `pending` will be carried to the next non-empty chunk